src/main/java/nl/ordina/jobcrawler/service/VacancyStarter.java

```diff
@@ -0,0 +1,88 @@
+package nl.ordina.jobcrawler.service;
+
+import nl.ordina.jobcrawler.controller.MylerVacancyScraper;
+import nl.ordina.jobcrawler.controller.YachtVacancyScraper;
+import nl.ordina.jobcrawler.model.Vacancy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.List;
+
+/*
+This 'starter' class uses the @Scheduled annotation. Every 15 minutes it executes the cronJobSch() function to retrieve all vacancies.
+Upon fetching the vacancies it runs a check to verify if the vacancy is already present in the database.
+*/
+
+@Component
+public class VacancyStarter {
+
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    private VacancyService vacancyService;
+
+    private final MylerVacancyScraper mylerVacancyScraper;
+    private final YachtVacancyScraper yachtVacancyScraper;
+
+    @Autowired
+    public VacancyStarter(MylerVacancyScraper mylerVacancyScraper, YachtVacancyScraper yachtVacancyScraper) {
+        this.mylerVacancyScraper = mylerVacancyScraper;
+        this.yachtVacancyScraper = yachtVacancyScraper;
+    }
+
+    @Scheduled(cron = "0 0/15 * * * *")
+    public void cronJobSch() throws IOException {
+        // This function gets executed every 15 minutes
+        log.info("CRON Scheduled");
+        scrapeYacht();
+        //Disbale Myler as it does not scrape entire vacancy yet.
+//        scrapeMyler();
+        System.out.println();
+    }
+
+    private void scrapeYacht() throws IOException {
```

Comment:
scrapeYacht en scrapeMyler zijn wel heel erg hetzelfde. Dit kan samengevoegd worden.

Reply:
Functies zijn samengevoegd in https://github.com/mostafa-221/jobcrawler-backend/commit/e88d1952efbd280688c2e3f895b8d232a2355b66